settings.lua

```diff
@@ -1551,7 +1544,129 @@ function IntSettingScientific:getFixedValue()
 	return self:get()*self.POWER
 end
 
+---@class DriverModeSetting : SettingList
+DriverModeSetting = CpObject(SettingList)
+DriverModeSetting.GRAIN_TRANSPORT = 1
+DriverModeSetting.COMBINE_UNLOAD = 2
+DriverModeSetting.OVERLOAD = 3
+DriverModeSetting.FILLABLE_FIELDWORK = 4
+DriverModeSetting.TRANSPORT = 5
+DriverModeSetting.FIELDWORK = 6
+DriverModeSetting.BALE_COLLECTOR = 7
+DriverModeSetting.FIELD_SUPPLY = 8
+DriverModeSetting.SHOVEL = 9
+DriverModeSetting.BUNKER_SILO = 10
+DriverModeSetting.NUM_MODES = 10
+
+DriverModeSetting.defaultMode = DriverModeSetting.TRANSPORT
+
+function DriverModeSetting:init(vehicle)
+	local values = {}
+	local texts = {}
+	for i=1,self.NUM_MODES do 
+		values[i] = i
+		texts[i] = i
+	end
+	SettingList.init(self,"driverMode","driverMode","driverMode",vehicle,values,texts)
+	self.current = self.defaultMode
+end
+
+function DriverModeSetting:setLoadedValue()
+
+end
+
+function DriverModeSetting:postInit()
+	self:set(self.valueFromXml,true)
+end
+
+
+function DriverModeSetting:validateCurrentValue()
+	self.validModes = {}
+	local wasResetToDefault = false
+	for i=1,self.NUM_MODES do 
+		local valid = courseplay:getIsToolCombiValidForCpMode(self.vehicle,i)
+		self.validModes[i] = valid or false
+		if self:get() == i and not valid then 
+			self:resetToDefault()
+			wasResetToDefault = true
+		end
+	end
+end
+
+function DriverModeSetting:onChange()
+	self:setAIDriver()
+	self.vehicle.cp.driver:refreshHUD()
+end
+
+function DriverModeSetting:setAIDriver()
+	courseplay.infoVehicle(self.vehicle, 'Setting mode %d', self:get())
+	courseplay:setAIDriver(self.vehicle, self:get())
+end
+
+function DriverModeSetting:resetToDefault()
+	self:set(self.defaultMode)
+end
+
+function DriverModeSetting:getDefaultMode()
+	return self.defaultMode
+end
+
+function DriverModeSetting:getValidModes()
+	return self.validModes
+end
+
+function DriverModeSetting:checkAndSetValidValue(new)
+	if courseplay:getIsToolCombiValidForCpMode(self.vehicle,new) then 
+		return SettingList.checkAndSetValidValue(self,new)
+	else 
+		return self.defaultMode
+	end
+end
+
+function DriverModeSetting:isDisabled()
+	return not self.vehicle.cp.canSwitchMode or self.vehicle:getIsCourseplayDriving() 
+end
+
+--- Set the next value
+function DriverModeSetting:setNext()
+	if not self:isDisabled() then
+		local new = self:getNextMode()
+		self:setToIx(new)
+	end
+end
+
+--- Set the previous value
+function DriverModeSetting:setPrevious()
+	if not self:isDisabled() then
+		local new = self:getPrevMode()
+		self:setToIx(new)
+	end
+end
 
+function DriverModeSetting:getNextMode()
```

Comment:
Why can't we use the setNext/setPrevious from SettingList? Would avoid this awkward loop (repeated twice)